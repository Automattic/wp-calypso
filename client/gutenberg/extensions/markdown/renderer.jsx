/** @format */

/**
 * External dependencies
 */
import MarkdownIt from 'markdown-it';
import { __ } from '@wordpress/i18n';
import { RawHTML } from '@wordpress/element';

/**
 * Module variables
 */
const markdownConverter = new MarkdownIt();
const handleLinkClick = event => {
	if ( event.target.nodeName === 'A' ) {
		const hasConfirmed = window.confirm(
			__( 'Are you sure you wish to leave this page?', 'jetpack' )
		);

		if ( ! hasConfirmed ) {
			event.preventDefault();
		}
	}
};

export default ( { className, source = '' } ) => (
	<RawHTML className={ className } onClick={ handleLinkClick }>
		{ source.length ? markdownConverter.render( source ) : '' }
	</RawHTML>
);
