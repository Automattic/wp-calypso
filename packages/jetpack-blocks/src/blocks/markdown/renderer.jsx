/**
 * External dependencies
 */
import MarkdownIt from 'markdown-it';
import { RawHTML } from '@wordpress/element';

/**
 * Internal dependencies
 */
import { __ } from '../../utils/i18n';

/**
 * Module variables
 */
const markdownConverter = new MarkdownIt();
const handleLinkClick = event => {
	if ( event.target.nodeName === 'A' ) {
		const hasConfirmed = window.confirm( __( 'Are you sure you wish to leave this page?' ) );

		if ( ! hasConfirmed ) {
			event.preventDefault();
		}
	}
};

export default ( { className, source = '' } ) => (
	<RawHTML className={ className } onClick={ handleLinkClick }>
		{ source.length ? markdownConverter.render( source ) : '' }
	</RawHTML>
);
