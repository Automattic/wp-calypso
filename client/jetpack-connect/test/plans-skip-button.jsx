/**
 * @format
 * @jest-environment jsdom
 */

/**
 * External dependencies
 */
import React from 'react';
import PlansSkipButton from 'client/jetpack-connect/plans-skip-button';
import renderer from 'react-test-renderer';

describe( 'PlansSkipButton', () => {
	test( 'should render', () => {
		const component = renderer.create( <PlansSkipButton /> );
		const tree = component.toJSON();
		expect( tree ).toMatchSnapshot();
	} );

	test( 'should render arrow-left in rtl mode', () => {
		const component = renderer.create( <PlansSkipButton isRtl /> );
		const tree = component.toJSON();
		expect( tree ).toMatchSnapshot();
	} );
} );
