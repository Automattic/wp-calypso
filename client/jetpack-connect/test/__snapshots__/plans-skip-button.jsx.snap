// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlansSkipButton should render 1`] = `
<div
  className="jetpack-connect__plans-nav-buttons"
>
  <button
    className="button"
    onClick={undefined}
    type="button"
  >
    Start with free
    <svg
      className="gridicon gridicons-arrow-right needs-offset-y"
      height={18}
      onClick={undefined}
      viewBox="0 0 24 24"
      width={18}
      xmlns="http://www.w3.org/2000/svg"
    >
      <g>
        <path
          d="M12 4l-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8-8-8z"
        />
      </g>
    </svg>
  </button>
</div>
`;

exports[`PlansSkipButton should render arrow-left in rtl mode 1`] = `
<div
  className="jetpack-connect__plans-nav-buttons"
>
  <button
    className="button"
    onClick={undefined}
    type="button"
  >
    Start with free
    <svg
      className="gridicon gridicons-arrow-left needs-offset-y"
      height={18}
      onClick={undefined}
      viewBox="0 0 24 24"
      width={18}
      xmlns="http://www.w3.org/2000/svg"
    >
      <g>
        <path
          d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
        />
      </g>
    </svg>
  </button>
</div>
`;
